### PR TITLE
KOGITO-2467: Use FailSafe maven plugin for Integration Tests

### PR DESCRIPTION
**Pull request by Sgitario**
JIRA: https://issues.redhat.com/browse/KOGITO-2467
Description:
- Use surefire for tests that do not need external components
- Use failsafe for tests that need external components like Kafka or Infinispan

Also, I've provided the persistence and events profiles in the parent as not all the examples have these profiles.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Review by sutaakar — APPROVED** [review]

**Review by cristianonicolai — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -133,53 +136,93 @@
       <url>http://kogito.kie.org/community/team.html</url>
     </developer>
   </developers>
-
+
```
Comment (jstastny-cz) [inline pom.xml:139]: trailing ws ;-)

**Review by jstastny-cz — COMMENTED** [review]
Why has `process-kafka-quickstart-springboot/src/test/java/org/acme/travel/MessagingIntegrationTest.java` been deleted?

**Comment by jstastny-cz** [comment]
@Sgitario by adding the profile persistence to parent pom.xml - isn't it possible now to run with the profile even in projects that are not in the `modules` section of the profile by going directly to the project and invoking with `-Ppersistence`?

File: pom.xml (hunk 1 above)
Comment (Sgitario) [reply pom.xml:139]: There is only one ws between </developers> and <modules>. I think it's fine.

**Comment by Sgitario** [comment]
> Why has `process-kafka-quickstart-springboot/src/test/java/org/acme/travel/MessagingIntegrationTest.java` been deleted?

It has been renamed to MessagingIT.java

**Comment by Sgitario** [comment]
> @Sgitario by adding the profile persistence to parent pom.xml - isn't it possible now to run with the profile even in projects that are not in the `modules` section of the profile by going directly to the project and invoking with `-Ppersistence`?

I can run "mvn clean install -Ppersistence" from the module path without failures. Maven will say that the profile does not exist and will use the default profile.

**Review by jstastny-cz — APPROVED** [review]

**Pull request merged**